public/app/features/plugins/built_in_plugins.ts

```diff
@@ -73,10 +73,23 @@ const canvasPanel = async () => await import(/* webpackChunkName: "canvasPanel"
 const iconPanel = async () => await import(/* webpackChunkName: "iconPanel" */ 'app/plugins/panel/icon/module');
 const graphPanel = async () => await import(/* webpackChunkName: "graphPlugin" */ 'app/plugins/panel/graph/module');
 const heatmapPanel = async () =>
-  await import(/* webpackChunkName: "heatmapPlugin" */ 'app/plugins/panel/heatmap/module');
+  await import(/* webpackChunkName: "heatmapPanel" */ 'app/plugins/panel/heatmap/module');
+const heatmapPanelOLD = async () =>
+  await import(/* webpackChunkName: "heatmapPanelOLD" */ 'app/plugins/panel/heatmap-old/module');
+
 const tableOldPanel = async () =>
   await import(/* webpackChunkName: "tableOldPlugin" */ 'app/plugins/panel/table-old/module');
 
+// Automatically migrate heatmap panel.
+if (config.featureToggles.useLegacyHeatmapPanel) {
+  const heatmap = config.panels['heatmap'];
+  const legacy = config.panels['heatmap-old'];
+  legacy.id = heatmap.id;
+  legacy.module = heatmap.module;
+  legacy.state = heatmap.state;
+  config.panels['heatmap'] = legacy;
```

Comment:
after this we will have two entires in config.panels with key heatmap and heatmap-old which have the same id (heatmap). Causes some issues in viz picker:  
![Screenshot from 2022-06-13 16-11-06](https://user-images.githubusercontent.com/10999/173373054-30aad171-6157-4706-8439-2cdf4b6b093f.png)
![Screenshot from 2022-06-13 16-06-53](https://user-images.githubusercontent.com/10999/173373072-fd9a5890-5658-46df-a774-00e151e2d2d9.png)

we could add `delete config['heatmap-old]'];`   seems to fix the issue